### PR TITLE
Run on later Chef versions

### PR DESCRIPTION
**Pull request by swalberg**
### Description

Trying to use this cookbook with later Chef releases but am running into
strange problems. Here I saved a node attribute that got re-used several
times as the existing method of calling seemed to overwrite the value so
`get_key` was getting an empty hash.

I'm not sure this case is fully exercised in the Test Kitchen suite. Also made a change to specify the version of the Chef Client.

### Issues Resolved

My wrapper cookbook was getting:

```       ================================================================================
       Recipe Compile Error in /tmp/kitchen/cache/cookbooks/nfl-dhcp/recipes/default.rb
       ================================================================================

       Chef::Exceptions::InvalidDataBagItemID
       --------------------------------------
       Data Bag items must have an id matching /^[\.\-[:alnum:]_]+$/, you gave: {}

       Cookbook Trace:
       ---------------
         /tmp/kitchen/cache/cookbooks/dhcp/libraries/dynadns.rb:93:in `get_key'
         /tmp/kitchen/cache/cookbooks/dhcp/libraries/dynadns.rb:76:in `keys'
         /tmp/kitchen/cache/cookbooks/dhcp/recipes/_config.rb:20:in `block in from_file'
         /tmp/kitchen/cache/cookbooks/dhcp/recipes/_config.rb:9:in `from_file'
         /tmp/kitchen/cache/cookbooks/dhcp/recipes/server.rb:26:in `from_file'
         /tmp/kitchen/cache/cookbooks/nfl-dhcp/recipes/default.rb:9:in `from_file'
```
### Contribution Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by swalberg** [comment]
Looking into those foodcritic errors... They are not related to my change. I'll have a look -- did you want those fixed in this PR?

```
sean~/work/dhcp (dyndnskeys *)$ foodcritic -f any . --exclude spec -t ~FC059
Checking 40 files
........xxxxx...........................
FC085: Resource using new_resource.updated_by_last_action to converge resource: ./providers/class.rb:19
FC085: Resource using new_resource.updated_by_last_action to converge resource: ./providers/group.rb:22
FC085: Resource using new_resource.updated_by_last_action to converge resource: ./providers/group.rb:32
FC085: Resource using new_resource.updated_by_last_action to converge resource: ./providers/host.rb:24
FC085: Resource using new_resource.updated_by_last_action to converge resource: ./providers/host.rb:33
FC085: Resource using new_resource.updated_by_last_action to converge resource: ./providers/shared_network.rb:39
FC085: Resource using new_resource.updated_by_last_action to converge resource: ./providers/shared_network.rb:50
FC085: Resource using new_resource.updated_by_last_action to converge resource: ./providers/subnet.rb:31
FC085: Resource using new_resource.updated_by_last_action to converge resource: ./providers/subnet.rb:42
```

**Comment by swalberg** [comment]
The test kitchen run works on all except ubuntu-1604.

```
Instance             Driver   Provisioner     Verifier  Transport  Last Action    Last Error
default-centos-68    Vagrant  PolicyfileZero  Inspec    Ssh        <Not Created>  <None>
default-centos-73    Vagrant  PolicyfileZero  Inspec    Ssh        <Not Created>  <None>
default-ubuntu-1404  Vagrant  PolicyfileZero  Inspec    Ssh        <Not Created>  <None>
default-ubuntu-1604  Vagrant  PolicyfileZero  Inspec    Ssh        Set Up         Kitchen::ActionFailed
```

File: libraries/dynadns.rb [hunk 1]
```diff
@@ -68,7 +68,8 @@ def keys
         # TODO: need to work out the namespace on dns stuff here.
         # TODO: be good to support knife-vault/encrypted bags for keys
         if node.key?(:dns) && node['dns'].key?(:rndc_key)
-          k[node.normal['dns']['rndc_key']] = get_key node['dns']['rndc_key']
+          default_key = node['dns']['rndc_key']
+          k[default_key] = get_key(default_key)
```
Comment (josephholsten) [inline libraries/dynadns.rb]: oh look, now I can actually read this code! :squee:

File: .kitchen.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ---
 driver:
   name: vagrant
-  chef_version: 12.19.36
+  require_chef_omnibus: 12.21.3
```
Comment (josephholsten) [inline .kitchen.yml]: any reason for this particular version?

Comment (swalberg) [reply .kitchen.yml]: It's what we've settled on at work until we're ready for Chef 13.

**Pull request merged**